frontend/src/components/ServiceForm/ServiceForm.tsx

```diff
@@ -45,6 +52,17 @@ export const ServiceForm: React.FC<Props> = ({ service, target = DEFAULT_TARGET,
   const appType = findType(applicationTypes, form.type)
   const css = useStyles()
 
+  useEffect(() => {
+    if ((form?.port, form?.hostname)) {
```

Comment:
I'm don't think you want the `,` comma here... maybe `&&` instead? Also let's discuss not scanning unless we have a valid hostname and port.